src/CodeGeneration.Roslyn/TransformationContext.cs

```diff
@@ -12,13 +17,19 @@ public class TransformationContext
         /// <param name="semanticModel">The semantic model.</param>
         /// <param name="compilation">The overall compilation being generated for.</param>
         /// <param name="projectDirectory">The absolute path of the directory where the project file is located</param>
+        /// <param name="compilationUnitUsings">The using directives already queued to be generated.</param>
+        /// <param name="compilationUnitExterns">The extern aliases already queued to be generated.</param>
         public TransformationContext(CSharpSyntaxNode processingNode, SemanticModel semanticModel, CSharpCompilation compilation,
-                                     string projectDirectory)
+                                     string projectDirectory,
+                                     IEnumerable<UsingDirectiveSyntax> compilationUnitUsings,
+                                     IEnumerable<ExternAliasDirectiveSyntax> compilationUnitExterns)
         {
```

Comment:
While this is a technically breaking change, it should only break unit tests that may have been written for generators, which isn't as severe.
We could make this internal, but that would make mocking up those unit tests even harder. So I'm good with this change.